espresso/streamer.go

```diff
@@ -323,6 +326,14 @@ func (s *EspressoStreamer[B]) HasNext(ctx context.Context) bool {
 
 // This function allows to "pin" the Espresso block height corresponding to the last safe batch
 // Note that this function can be called
-func (s *EspressoStreamer[B]) confirmEspressoBlockHeight() {
-	s.confirmedHotShotPos = s.hotShotPos
+func (s *EspressoStreamer[B]) confirmEspressoBlockHeight(safeL1Origin eth.BlockID) error {
+	hotshotState, err := s.EspressoLightClient.
+		FinalizedState(&bind.CallOpts{BlockNumber: new(big.Int).SetUint64(safeL1Origin.Number)})
+	if err != nil {
+		return err
+	}
+
+	// TODO(AG): this is not correct and causes 8.1.2. to fail
```

Comment:
Is this TODO still relevant as the test 8.1.2 is passing?

Reply:
No, removed